controllers/subscription/subscription_controller.go

```diff
@@ -248,7 +252,15 @@ func (r *SubscriptionReconciler) HandleUpgrades(ctx context.Context, rhmiSubscri
 		return ctrl.Result{}, nil
 	}
 
-	if !isServiceAffecting && !latestInstallPlan.Spec.Approved {
+	upgradeAllowed, err := r.upgradeAllowed()
+	if err != nil {
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: 10 * time.Second,
```

Comment:
In the case of blocking the upgrade, do we want to Requeue it this frequently? Will this run ever 10 seconds until the config map is removed?

Reply:
no, controller will be re-queued only if there's an error fetching the cm which is not isNotFound error, for example, temporary connection error